File: app/src/main/java/com/nithinkumar/water/Activity/WaterActivity.java
package com.nithinkumar.water.Activity;

import android.os.Bundle;
import android.view.MenuItem;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.nithinkumar.water.Fragment.OnboardingFragment;
import com.nithinkumar.water.R;
import com.nithinkumar.water.WaterShared;
import com.nithinkumar.water.fragment.DetailsDialogFragment;
import com.nithinkumar.water.fragment.HomeFragment;
import com.nithinkumar.water.fragment.SettingsFragment;
import com.nithinkumar.water.fragment.WaterIntakeFragment;

public class WaterActivity extends AppCompatActivity {

    FragmentManager fragmentManager;
    FragmentTransaction fragmentTransaction;
    WaterShared mWaterShared;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_water);
        mWaterShared = new WaterShared(this);

        BottomNavigationView navigation = findViewById(R.id.navigation);
        fragmentManager = getSupportFragmentManager();

        if (!mWaterShared.getSeenOnboarding()) {
            goTo(OnboardingFragment.Companion.newInstance());
            mWaterShared.setSeenOnboarding(true);
        } else if (mWaterShared.getFirstRun().equals("")) {
            DetailsDialogFragment ddf = new DetailsDialogFragment();
            ddf.show(fragmentManager, "ActivityDialog");
        } else {
            mWaterShared.setFirstRun("Done");
            //afterCheck();
        }

        afterCheck();
        navigation.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener);
    }

    private void goTo(Fragment fragment) {
        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
        fragmentTransaction.add(R.id.content, fragment);
        fragmentTransaction.addToBackStack("GoTo " + fragment.getTag());
        fragmentTransaction.commit();
    }

    private BottomNavigationView.OnNavigationItemSelectedListener mOnNavigationItemSelectedListener
            = new BottomNavigationView.OnNavigationItemSelectedListener() {

        @Override
        public boolean onNavigationItemSelected(@NonNull MenuItem item) {
            switch (item.getItemId()) {
                case R.id.navigation_home:
                    fragmentTransaction = fragmentManager.beginTransaction();
                    fragmentTransaction.replace(R.id.content, HomeFragment.newInstance(R.layout.fragment_home), "1");
                    fragmentTransaction.commit();
                    return true;
                case R.id.navigation_dashboard:
                    fragmentTransaction = fragmentManager.beginTransaction();
                    fragmentTransaction.replace(R.id.content, WaterIntakeFragment.newInstance(), "2");
                    fragmentTransaction.commit();
                    return true;
                case R.id.navigation_notifications:
                    fragmentTransaction = fragmentManager.beginTransaction();
                    fragmentTransaction.replace(R.id.content, SettingsFragment.newInstance(R.layout.fragment_settings), "3");
                    fragmentTransaction.commit();
                    return true;
            }
            return false;
        }

    };

    private void afterCheck() {

        fragmentTransaction = fragmentManager.beginTransaction();
        fragmentTransaction.add(R.id.content, HomeFragment.newInstance(R.layout.fragment_home), "1");
        fragmentTransaction.commit();
    }

}

File: app/src/main/java/com/nithinkumar/water/Activity/WaterActivity.kt
package com.nithinkumar.water.activity

import android.app.ProgressDialog.show
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.FragmentManager
import com.nithinkumar.water.R
import com.nithinkumar.water.WaterShared
import com.nithinkumar.water.fragment.*

class WaterActivity : AppCompatActivity(),
        SplashScreenFragment.OnSplashScreenFragmentInteractionListener,
        WaterFragment.OnWaterScreenFragmentInteractionListener,
        HomeFragment.OnHomeScreenFragmentInteractionListener {

    private val fragmentManager: FragmentManager = supportFragmentManager
    private lateinit var mWaterShared: WaterShared

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_water)
        mWaterShared = WaterShared(this)
        initialLoad()
    }

    override fun redirectToHomeScreen() {
        fragmentManager.popBackStack()
        fragmentManager.beginTransaction().apply {
            replace(R.id.content, WaterFragment.newInstance(), "2")
            commit()
        }
    }

    private fun initialLoad() {
        fragmentManager.beginTransaction().apply {
            add(R.id.content, SplashScreenFragment.newInstance(), "1")
            addToBackStack(null)
            commit()
        }
    }

    override fun onDefaultLoad() {
        fragmentManager.beginTransaction().apply {
            add(R.id.pageContainer, HomeFragment.newInstance(), "3")
            commit()
        }
    }

    override fun onHomeClicked() {
        fragmentManager.beginTransaction().apply {
            replace(R.id.pageContainer, HomeFragment.newInstance(), "3")
            commit()
        }

    }

    override fun onHistoryClicked() {
        fragmentManager.beginTransaction().apply {
            replace(R.id.pageContainer, HistoryFragment.newInstance(), "4")
            commit()
        }

    }

    override fun onSettingsClicked() {
        fragmentManager.beginTransaction().apply {
            replace(R.id.pageContainer, SettingsFragment.newInstance(), "5")
            commit()
        }
    }

    override fun onFABClicked() {
        WaterIntakeFragment().let { addWater ->
            addWater.setTargetFragment(fragmentManager.findFragmentByTag("2"), 280)
            fragmentManager.let { addWater.show(it, "Quantity Consumed") }
        }
    }
}
